influxdb3_wal/src/object_store.rs

```diff
@@ -743,6 +836,32 @@ impl<'a> TryFrom<&'a Path> for WalFileSequenceNumber {
     }
 }
 
+#[derive(Debug)]
+struct WalFileRemover {
+    oldest_wal_file: Option<WalFileSequenceNumber>,
+    last_wal_sequence_number: Option<WalFileSequenceNumber>,
```

Comment:
rename to `last_snapshotted_wal_sequence_number` for clarity. The latest wal file number will always be >= this one.